packages/teams-js/src/public/pages.ts

```diff
@@ -674,9 +832,27 @@ export namespace pages {
      * @param handler - The handler to invoke when exiting hover of the personal app button in the app bar.
      */
     export function onHoverLeave(handler: () => void): void {
+      onHoverLeaveHelper(handler, () => {
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
+      });
+    }
+
+    /**
+     * @hidden
+     * Undocumented helper function with shared code between deprecated version and current version of the onHoverLeave API.
+     *
+     * @internal
+     * Limited to Microsoft-internal use
+     *
+     * @param handler - The handler to invoke when existing hover of the personal app button in the app bar.
+     * @param versionSpecificHelper - The helper function containing logic pertaining to a specific version of the API.
+     */
+    export function onHoverLeaveHelper(handler: () => void, versionSpecificHelper?: () => void): void {
```

Comment:
All of these helpers look very similar. Is there some way we can have a single shared helper function for this case? Maybe taking in the message name, a generic handler object, and the versionSpecificHelper closure?

Reply:
I have gone over this. There isn't a way to do this without leaving _some_ duplicated code present in the library. Either we have duplicate handler helpers (as it is currently) or we have one shared helper function and duplicated code in both v1 and v2 versions of each handler that sets up the `ensureInitialized` call (since that is different across the register*Handler calls), and duplicated code to pass the name of the handler around. 

My preference given that this needs to get into the next release is to leave it as-is. If anyone feels strongly that they would prefer the duplicated code in each of the v1/v2 versions of the register*Handler calls, I will open a work item to track fixing it post-next-release.

Reply:
I think it's fine to leave as-is: if it becomes a problem we can revisit in the future. We've talked about it offline and I recognize that this is a slightly trickier problem than we thought.

fwiw, I don't think having duplicated code between v1 and v2 is bad: I think actually pulling apart v1 and v2 functions more over time might help not introduce bugs like this